samza-api/src/main/java/org/apache/samza/runtime/ApplicationStatus.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.runtime;
+
+public class ApplicationStatus {
```

Comment:
So after this PR and @navina change, we're going to have 
JobStatus
ApplicationStatus
ContainerStatus
... others?

It'd be nice to have a clear purpose for each of these. 

For example, I don't see why ApplicationStatus needs to handle exceptions, but JobStatus doesn't. If anything, I'd think the only difference between the two is that ApplicationStatus spans multiple jobs, so it might have a status to indicate PartialFinish for the case that a subset of the jobs are not running. Otherwise, I don't see why both are needed.